### PR TITLE
Add method 'posix-real' to show posix value with fractional seconds

### PR DESCRIPTION
**Pull request by tbrowder**

### CODE REVIEW

**Review by vrurg — CHANGES_REQUESTED** [review]

File: src/core.c/DateTime.pm6 [hunk 1]
```diff
@@ -414,6 +414,11 @@ my class DateTime does Dateish {
         self.modified-julian-date + 2_400_000.5
     }
 
+    method posix-real(DateTime:D: $ignore-timezone? --> Real:D) {
+	    my $p = $ignore-timezone ?? self.posix($ignore-timezone) !! self.posix;
```
[the remark this answers is not in the document]
Comment (tbrowder) [reply src/core.c/DateTime.pm6:418]: Okay, sounds good. I wasn't sure about how to handle the optional flag on $ignore-timezone.

File: src/core.c/DateTime.pm6 [hunk 2]
```diff
@@ -414,6 +414,11 @@ my class DateTime does Dateish {
...
+	    $p + self.second - self.second.Int
```
[the remark this answers is not in the document]
Comment (tbrowder) [reply src/core.c/DateTime.pm6:419]: Also sounds good.

**Comment by tbrowder** [comment]
@vrurg, that is, sorry

**Review by vrurg — APPROVED** [review]

**Comment by vrurg** [comment]
@tbrowder don't be, it's a normal work process. 🙂

Thank you!

**Comment by vrurg** [comment]
I wouldn't merge this PR because I'm still not convinced that there is no better way to have all this functionality added but via piling up new methods.

**Comment by tbrowder** [comment]
I did get one up-vote on the original issue about needing a real value for the existng 'posix' method. I did look into making the existing 'posix' method yield a real value, but it seemed to affect too much other core code (including nqp); but that is what I wish could be done. It makes no sense to me for it to be an integral value facing the user.

I assume that 'posix' was made an integer for speed, but changing it to another name for internal use and make 'posix' real is my suggestion.

**Comment by lizmat** [comment]
I think it's more for historical reasons, and compatibility with `time`, which is also an `Int`.

**Comment by lizmat** [comment]
Ok, looking at this again now, and the actual `DateTime.posix` code, I think the best way to do this, would be to add a `:real` named argument to `.posix`.  @tbrowder would that work for you?

**Comment by vrurg** [comment]
I was considering a named too, but `posix` is `--> Int:D`. It can be `--> Real:D` as well, but we'd need to do a blin run to make sure it breaks nothing.

**Comment by tbrowder** [comment]
> Ok, looking at this again now, and the actual `DateTime.posix` code, I think the best way to do this, would be to add a `:real` named argument to `.posix`. @tbrowder would that work for you?

Good idea! I think so, let me think about it for a bit.

**Comment by lizmat** [comment]
@tbrowder hold your horses  :-)

How about we make it even easier and change the semantics of `DateTime.posix` for 6.e to return the real value?  That would allow people to opt-in.

**Comment by tbrowder** [comment]
Superseded by new PR #4371

**Pull request closed without merging**